dev-tools/openapi-manager/src/check.rs

```diff
@@ -59,56 +59,85 @@ pub(crate) fn check_impl(
+                    // Success case.
+                    let extra = if status.extra_files_len() > 0 {
+                        format!(
+                            ", {} extra files",
+                            status.extra_files_len().style(styles.bold)
+                        )
+                    } else {
+                        "".to_string()
+                    };
 
-                    num_up_to_date += 1;
-                }
-                CheckStatus::Stale { full_path, actual, expected } => {
                     eprintln!(
-                        "{:>HEADER_WIDTH$} [{count:>count_width$}/{total}] {}",
-                        STALE.style(styles.warning_header),
+                        "{:>HEADER_WIDTH$} [{count:>count_width$}/{total}] {}: {}{extra}",
+                        UP_TO_DATE.style(styles.success_header),
                         display_api_spec(spec, &styles),
+                        display_summary(&status.summary, &styles),
                     );
 
-                    let diff = TextDiff::from_lines(&actual, &expected);
-                    write_diff(
-                        &diff,
-                        &full_path,
-                        &styles,
-                        // Add an indent to align diff with the status message.
-                        &mut IndentWriter::new(
-                            &continued_indent,
-                            std::io::stderr(),
-                        ),
-                    )?;
-
-                    num_stale += 1;
+                    num_up_to_date += 1;
+                    continue;
                 }
-                CheckStatus::Missing => {
-                    eprintln!(
-                        "{:>HEADER_WIDTH$} [{count:>count_width$}/{total}] {}",
-                        MISSING.style(styles.warning_header),
-                        display_api_spec(spec, &styles),
-                    );
 
-                    num_missing += 1;
+                // Out of date: print errors.
+                eprintln!(
+                    "{:>HEADER_WIDTH$} [{count:>count_width$}/{total}] {}",
+                    OUT_OF_DATE.style(styles.warning_header),
+                    display_api_spec(spec, &styles),
+                );
+                num_out_of_date += 1;
+
+                for (error_ix, (spec_file, error)) in
+                    status.iter_errors().enumerate()
+                {
+                    let error_count = error_ix + 1;
+
+                    match error {
+                        CheckError::Stale { full_path, actual, expected } => {
+                            eprintln!(
+                                "{:>HEADER_WIDTH$}{count_section_indent}\
+                                 ({error_count:>total_errors_width$}/{total_errors}) {}",
+                                STALE.style(styles.warning_header),
+                                display_api_spec_file(spec, spec_file, &styles),
+                            );
```

Comment:
this is _almost_ identical to the code in the missing case. i wonder if we could have the `CheckError` type return which header should be used, or something, and then have one copy of the format string?

this is very much not a big deal, i just go out of my way to avoid duplicating identical format strings, to make it easier to avoid accidentally changing the format in all cases except for one. up to you whether it's worth the effort.

Reply:
Thanks -- moved this into a callback to share code.